grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/RedisIndexStorageTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.test.rule.InMemoryRedisContext;
+import com.codahale.metrics.MetricRegistry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/**
+ * <p>
+ *    Tests that index related post processing data can be stored and retrieved from redis
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisIndexStorageTest {
+    @ClassRule
+    public static final InMemoryRedisContext IN_MEMORY_REDIS_CONTEXT = InMemoryRedisContext.create();
+
+    private static RedisIndexStorage indexStorage;
+    private final static Keyspace keyspace1 = Keyspace.of("myhappypotato");
+    private final static Keyspace keyspace2 = Keyspace.of("mysadapple");
+    private final static String index1 = "index1";
+    private final static String index2 = "index2";
+    private final static Set<ConceptId> conceptIds = new HashSet<>(Arrays.asList(ConceptId.of("a"), ConceptId.of("b"), ConceptId.of("c")));
+
+    @BeforeClass
+    public static void getConnection(){
+        indexStorage = RedisIndexStorage.create(IN_MEMORY_REDIS_CONTEXT.jedisPool(), new MetricRegistry());
+    }
+
+    @Before
+    public void addDataToStorage(){
+        indexStorage.addIndex(keyspace1, index1, conceptIds);
+        indexStorage.addIndex(keyspace2, index1, conceptIds);
+        indexStorage.addIndex(keyspace2, index2, conceptIds);
+    }
+
+    @After
+    public void removeDataFromStorage(){
+        indexStorage.popIndex(keyspace1);
+        indexStorage.popIndex(keyspace2);
+        indexStorage.popIds(keyspace1, index1);
+        indexStorage.popIds(keyspace2, index1);
+        indexStorage.popIds(keyspace2, index2);
+    }
+
+    @Test
+    public void whenAddingIndices_EnsureKeyspaceListContainIndices(){
```

Comment:
perhaps more aptly named `onPreinitializedIndexStorage_ensureKeyspaceListContainIndices` (or something along those lines). As we're adding the indices during the `@Before` step instead of on the test itself.

Reply:
Yar! Agreed!